homeassistant/components/obihai/config_flow.py

```diff
@@ -63,6 +76,9 @@ async def async_step_user(
     async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
         """Handle a flow initialized by importing a config."""
         self._async_abort_entries_match({CONF_HOST: config[CONF_HOST]})
+        if await async_credential_errors(self.hass, config):
+            return self.async_abort(reason="cannot_connect")
```

Comment:
I'm still not a great fan of creating the error dictionary inside the function.
I would still prefer something like this:
```python
    if await async_credential_errors(self.hass, user_input):
        return self.async_create_entry(...)
    return self.async_abort(reason="cannot_connect")
```

Reply:
Alright, I can live with this.